File: recipes-app/recipes-angularapp/src/app/classes/ingredient.ts
export class Ingredient {
    
    constructor(
        private _product: string,
        private _quantity: string
    ) {}

    public get quantity(): string {
        return this._quantity;
    }
    public set quantity(value: string) {
        this._quantity = value;
    }
    public get product(): string {
        return this._product;
    }
    public set product(value: string) {
        this._product = value;
    }
}

File: recipes-service/src/main/java/com/ironhack/recipesservice/classes/Ingredient.java
package com.ironhack.recipesservice.classes;

public class Ingredient {
    private String product;
    // quantity is a string because the metric differs between foods
    private String quantity;

    public Ingredient() {
    }

    public Ingredient(String product, String quantity) {
        this.product = product;
        this.quantity = quantity;
    }

    public String getProduct() {
        return product;
    }

    public void setProduct(String product) {
        this.product = product;
    }

    public String getQuantity() {
        return quantity;
    }

    public void setQuantity(String quantity) {
        this.quantity = quantity;
    }
}
